app/models/short_url_request.rb

```diff
@@ -25,6 +25,16 @@ class ShortUrlRequest
 
   scope :pending, -> { where(state: "pending") }
 
+  attr_accessor :confirmed
+
+  def duplicates
+    @duplicates ||= Redirect.or({from_path: from_path}, {to_path: to_path})
```

Comment:
How about `similar_redirects` ? I was planning on adding the "full duplicates" story as a simple validator method separate to these.